src/photos/ducks/timeline-clusters/index.jsx

```diff
@@ -39,20 +44,54 @@ const TIMELINE_MUTATIONS = client => ({
     })
 })
 
-const formatDate = (date, f) => {
-  return f(date, 'DD MMMM YYYY')
-}
-
+/**
+ *  Create the title for the section, following these rules:
+ *  - If the period is within same month: D - D M Y
+ *  - Else: D M Y - D M Y
+ *  - Exception: don't show the year if it is the current one.
+ */
 const getSectionTitle = (album, f) => {
   if (album.period) {
     const startPeriod = new Date(album.period.start)
     const endPeriod = new Date(album.period.end)
+
+    if (differenceInCalendarMonths(endPeriod, startPeriod) > 0) {
+      return formatDMY(f, startPeriod) + ' - ' + formatDMY(f, endPeriod)
+    }
     if (differenceInCalendarDays(endPeriod, startPeriod) > 0) {
-      // TODO Better period display
-      return formatDate(startPeriod, f) + ' - ' + formatDate(endPeriod, f)
+      return formatD(f, startPeriod) + '-' + formatDMY(f, endPeriod)
     }
+    return formatDMY(f, startPeriod)
   }
-  return formatDate(new Date(album.name), f)
+  return formatDMY(f, new Date(album.name))
```

Comment:
The cluster title will be `Invalid Date`. I could add a [validation check](https://date-fns.org/v1.30.1/docs/isValid) to avoid this and return an empty string instead.

Reply:
Yeah I think a default title other than `Invalid Date` would be good then :p Maybe we can even find something better than an empty string? But I'm not sure in what situation we could end up like this so it's hard to say what a pertinent default title would be.

Reply:
I was thinking about a solution using the current timeline behaviour in [PhotoBoard](https://github.com/cozy/cozy-drive/blob/master/src/photos/components/PhotoBoard.jsx#L57), but this could have some side-effects, if several photos are on the same day.

So instead, we could check the date validity [here](https://github.com/cozy/cozy-drive/blob/6f2397d2d15a30cdd6d5e7710f7e2e9d47c43dc3/src/photos/ducks/timeline-clusters/index.jsx#L137) and, if it is not valid, push the photo to `photosNotClustered` to force a clustering per day. 

What do you think?

Reply:
i'm still not sure in what situation this error can happen so it's hard to say. 

>  push the photo to photosNotClustered to force a clustering per day.

Aren't we looking into an album here? So you mean we would push all the photos il that album to `photosNotClustered ` ?